1. Added support for collapse field parameter for process search job

vsvvj-cb opened the pull request:
## Pull request checklist

Please check if your PR fulfills the following requirements:
<!-- These checkboxes can be checked like this: [x] no spaces between the brackets and the x!-->
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [ ] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [ ] A self-review of the code has been done.

## What is the ticket or issue number?
https://jira.carbonblack.local/browse/PRDTEST-15265


## Pull Request Description
An additional support is added for collapse_field parameter in payload for process search job API.


## Does this introduce a breaking change?

- [ ] Yes
- [x ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## How Has This Been Tested?
This is tested with E2E automation work flows.

avanbrunt-cb reviewed (changes requested):
Please add a test case which covers this new function. It can follow similar to the following test https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/tests/unit/platform/test_platform_process.py#L747

vsvvj-cb commented:
Added unit test case

> Please add a test case which covers this new function. It can follow similar to the following test https://github.com/carbonblack/carbon-black-cloud-sdk-python/blob/develop/src/tests/unit/platform/test_platform_process.py#L747

Added unit test case for this function.
